modules/core/src/main/java/org/apache/synapse/message/senders/blocking/BlockingMsgSenderUtils.java

```diff
@@ -240,8 +240,10 @@ public static void fillMessageContext(EndpointDefinition endpoint,
         if (endpoint.getEffectiveTimeout() > 0) {
             if (!endpoint.isDynamicTimeoutEndpoint()) {
                 axisOutMsgCtx.setProperty(SynapseConstants.SEND_TIMEOUT, endpoint.getEffectiveTimeout());
+                axisOutMsgCtx.setProperty(HTTPConstants.SO_TIMEOUT, (int) endpoint.getEffectiveTimeout());
             } else {
                 axisOutMsgCtx.setProperty(SynapseConstants.SEND_TIMEOUT, endpoint.evaluateDynamicEndpointTimeout(synapseInMsgCtx));
```

Comment:
Shall we define a local variable to hold the value of endpoint.evaluateDynamicEndpointTimeout(synapseInMsgCtx) ? 
Calling it twice might have a little performance impact.